### DISCUSSION TITLE
ERROR: onedrive application is already running - check system process list for active application instances

### DISCUSSION THREAD

**Discussion by hmzeahmd** [abraunegg/onedrive#2296] [General]
First you kill onedrive in "system monitor", is already running. that is the problem.
and then go to terminal

type onedrive --synchronize

**Comment by abraunegg** [comment]
@hmzeahmd 
> First you kill onedrive in "system monitor", is already running. that is the problem.

The 'problem' is actually that you have a running systemd service that is running the application already ... this is a *known* Ubuntu package bug.

To correctly stop the service - you need to correctly stop the systemd service .. not blindly kill the application - as systemd will just start it up again.

**Comment by ahagele** [comment]
I have the same problem and can't solve it.
So far I found out:

A second service wants to start up every few seconds - see syslog below.

At bootup an instance of onedrive.service is loaded but it's the 'wrong' one.
With 'systemctl list-units --type=service' I see 'onedrive@andreas.service'

If I manually 'end' this process, the 'right' one starts up automatically. It shows up in the process list as onedrive but the above 'systemctl list-units --type=service' doesn't show it (???)
After this 'end' all works fine and no further issue - until reboot, then I get the 'wrong' one loaded again. 
How can I find how and where this 'onedrive@andreas.service' is loaded?


```
Jun 12 10:28:45 Desktop8 onedrive[6402]: Configuring Global Azure AD Endpoints
Jun 12 10:28:45 Desktop8 onedrive[6402]: ERROR: onedrive application is already running - check system process list for active application instances
Jun 12 10:28:45 Desktop8 systemd[1438]: onedrive.service: Main process exited, code=exited, status=1/FAILURE
Jun 12 10:28:45 Desktop8 systemd[1438]: onedrive.service: Failed with result 'exit-code'.
Jun 12 10:28:48 Desktop8 onedrive[1323]: Got termination signal, performing clean up
Jun 12 10:28:48 Desktop8 onedrive[1323]: Shutting down the HTTP instance
Jun 12 10:28:48 Desktop8 onedrive[1323]: Shutting down db connection and merging temporary data
Jun 12 10:28:48 Desktop8 systemd[1]: onedrive@andreas.service: Deactivated successfully.
Jun 12 10:28:48 Desktop8 systemd[1438]: onedrive.service: Scheduled restart job, restart counter is at 28.
Jun 12 10:28:48 Desktop8 systemd[1438]: Stopped OneDrive Free Client.
Jun 12 10:28:48 Desktop8 systemd[1438]: Started OneDrive Free Client.
Jun 12 10:28:48 Desktop8 onedrive[6537]: onedrive.service: ProtectHostname=yes is configured, but UTS namespace setup is prohibited (container manager?), ignoring namespace setup.
Jun 12 10:28:49 Desktop8 onedrive[6537]: Configuring Global Azure AD Endpoints
Jun 12 10:28:51 Desktop8 onedrive[6537]: Initializing the Synchronization Engine ...
Jun 12 10:28:51 Desktop8 onedrive[6537]: Initializing monitor ...
Jun 12 10:28:51 Desktop8 onedrive[6537]: OneDrive monitor interval (seconds): 300
Jun 12 10:28:51 Desktop8 onedrive[6537]: Starting a sync with OneDrive
Jun 12 10:28:51 Desktop8 onedrive[6537]: Syncing changes and items from OneDrive ...
Jun 12 10:28:52 Desktop8 onedrive[6537]: Performing a database consistency and integrity check on locally stored data due to fullscan requirement ...
Jun 12 10:28:53 Desktop8 onedrive[6537]: Sync with OneDrive is complete
```

**Reply by ahagele** [reply]
Sorry that log clip formatting failed.
Got it attached now.
[syslogTail.txt](https://github.com/abraunegg/onedrive/files/11715933/syslogTail.txt)

**Reply by abraunegg** [reply]
To correctly quote logfiles use 3 quotation marks rather than a single one

**Comment by abraunegg** [comment]
@ahagele 
You essentially have probably installed a package that created double systemd entry files. You need to clean up your systemd entries.

Ubuntu and its clones have a bad habit of creating a 'default' systemd service file when installing the 'onedrive' package so that the client will automatically run the client post being authenticated. This systemd entry is erroneous and needs to be removed.
```
Created symlink /etc/systemd/user/default.target.wants/onedrive.service → /usr/lib/systemd/user/onedrive.service.
```
To remove this symbolic link, run the following command:
```
sudo rm /etc/systemd/user/default.target.wants/onedrive.service
```

**Comment by ahagele** [comment]
@abraunegg 
Thanks for the super quick reply.
I thought myself it's more a system issue than a onedrive problem.
BTW. great piece of software, works really well.

That symlink onedrive.service does not exist on my system (Mint 21) in that location. Nothing anywhere in /etc/systemd/* 
It must be hiding somewhere else.

**Comment by nirnachmani** [comment]
I am having exactly the same problem is @abraunegg. systemd keeps trying to start the onedrive service but fails because it is already running. If I kill the onedrive process it then succeeds and stop giving an error.  

However, I have no onedrive.service under /etc/systemd/user/default.target.wants/

I do have two service files for onedrive:

/usr/lib/systemd/user/onedrive.service
/usr/lib/systemd/system/onedrive@.service

My understanding is that the service with the "@" overrides the other one when it runs but is it possible that having both is causing the problem?

**Reply by abraunegg** [reply]
@nirnachmani 
Correct .. you cannot have these 2 service files active ... pick one .. you cannot run both

**Reply by nirnachmani** [reply]
Thanks!
